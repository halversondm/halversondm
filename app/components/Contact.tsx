/**
 * Created by Daniel on 6/26/2016.
 */
import * as React from "react";
import { type ReactNode, useState } from "react";
import { LinkedInButton } from "./LinkedInButton";
import { TwitterFollowButton } from "./TwitterFollowButton";

interface OwnState {
  subject: string;
  message: string;
}

export default function Contact(): ReactNode {
  const [state, setState] = useState<OwnState>(initialState());

  function initialState(): OwnState {
    return { subject: "", message: "" };
  }

  function captureSubject(event): void {
    const subject: string = event.target.value;
    setState({ ...state, subject });
  }

  function captureMessage(event): void {
    const message: string = event.target.value;
    setState({ ...state, message });
  }

  const href =
    "mailto:daniel.m.halverson@gmail.com" +
    "?subject=" +
    state.subject +
    "&body=" +
    state.message;
  return (
    <div>
      <h2 className="text-primary">Contact Information</h2>
      <div className="row">
        <div className="col-sm-4">
          <table className="table table-striped">
            <tbody>
              <tr>
                <td>Email</td>
                <td>
                  <a href="mailto:daniel.m.halverson@gmail.com">
                    daniel.m.halverson@gmail.com
                  </a>
                </td>
              </tr>
              <tr>
                <td>Twitter</td>
                <td>
                  <TwitterFollowButton />
                </td>
              </tr>
              <tr>
                <td>Linkedin</td>
                <td>
                  <LinkedInButton />
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <hr />
      <h4 className="text-success">Send me an email</h4>
      <form className="form">
        <input
          type="text"
          className="form-control"
          id="subject"
          onBlur={captureSubject}
          placeholder="Subject"
        />
        <br />
        <textarea
          id="message"
          onBlur={captureMessage}
          className="form-control"
          rows={3}
          placeholder="Message"
        />
        <br />
        <a className="btn btn-success" href={href}>
          Send
        </a>
      </form>
    </div>
  );
}
