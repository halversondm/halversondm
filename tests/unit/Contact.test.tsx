import React from "react";
import { render } from '@testing-library/react';
import Contact from "../../app/components/Contact";

test("Contact component", () => {
    const unit = <Contact/>;
    const component = render(unit);
    const tree = component.container;
    expect(tree.toString()).toMatchSnapshot();
});