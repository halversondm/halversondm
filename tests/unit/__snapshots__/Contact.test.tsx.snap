// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Contact component 1`] = `"<div><div><h2 class="text-primary">Contact Information</h2><div class="row"><div class="col-sm-4"><table class="table table-striped"><tbody><tr><td>Email</td><td><a href="mailto:daniel.m.halverson@gmail.com">daniel.m.halverson@gmail.com</a></td></tr><tr><td>Twitter</td><td><a href="https://twitter.com/halversondm" class="twitter-follow-button" data-show-count="true">Follow @halversondm</a></td></tr><tr><td>Linkedin</td><td><div class="badge-base LI-profile-badge" data-locale="en_US" data-size="medium" data-theme="light" data-type="VERTICAL" data-vanity="dmhalverson" data-version="v1"><a class="badge-base__link LI-simple-link" href="https://www.linkedin.com/in/dmhalverson?trk=profile-badge">Daniel Halverson</a></div></td></tr></tbody></table></div></div><hr><h4 class="text-success">Send me an email</h4><form class="form"><input type="text" class="form-control" id="subject" placeholder="Subject"><br><textarea id="message" class="form-control" rows="3" placeholder="Message"></textarea><br><a class="btn btn-success" href="mailto:daniel.m.halverson@gmail.com?subject=&amp;body=">Send</a></form></div></div>"`;
